pyproject.toml

```diff
@@ -0,0 +1,23 @@
+[tool.poetry]
+name = "data-source-guide-example"
+version = "0.1.0"
+description = ""
+authors = ["Andrew <dyusha.ru@gmail.com>"]
+readme = "README.md"
+
+[tool.poetry.dependencies]
+python = "^3.10"
+fastapi = "^0.115.12"
+uvicorn = "^0.34.0"
+
+
+[tool.poetry.group.dev.dependencies]
+invoke = "^2.2.0"
```

Comment:
Invoke should not be a dev-dependency. In order to run the app you need it. So it should be a normal dependency.

However I would actually suggest to remove it from the dependencies altogether and instead use typer. You should have access to the private repo https://github.com/ioxio-dataspace/data-sources and be able to check how it's done there. In short, you've got the dev/serve stuff in `[tool.poetry.scripts]` in this file (and `typer` as a dependency) and then in the `tasks` folder `dev.py` and `serve.py` + a `common.py`. Note that this change will then also require you to update the instructions in the `README.md` on how to run it (see the other repo for example).

Reply:
P.S. Hiding the stuff in the `tasks` folder likely makes it easier to spot the relevant stuff easily in this repo.